### ISSUE TITLE
Do we need controller.compile?

### ISSUE THREAD

**Issue by hoangcuong2011** [flomlo/ntm_keras#7]
Hi Florian,

Thanks for your great code! I have a question though:
Do we really need this line of code (line 81):

controller.compile(loss='binary_crossentropy', optimizer=sgd, metrics = ['binary_accuracy'], sample_weight_mode="temporal")

from https://github.com/flomlo/ntm_keras/blob/master/main.py ?

I don't really think we need this. If we do need this, it would really confuse me. We already specify (line 43):

    model.compile(loss='binary_crossentropy', optimizer=sgd, metrics = ['binary_accuracy'], sample_weight_mode="temporal")

from https://github.com/flomlo/ntm_keras/blob/master/model_ntm.py

controller is just a part of the NTM, and we should specific the ``compile" part only at NTM model. Am I correct?

I hope I made my question clear. Again, thanks a lot for your great work!

Best,

Cuong

**Comment by flomlo** [comment]
Hello and thank you for your interest!

I think that back then, when I wrote the code, it was necessary as
compiling the controller allowed to refer to something ... input/output
of the controller, maybe? Or some internal state? I dont remember
anymore, havent touched neither this code nor keras for almost a year.

I am also not sure if compiling is "recursive" in this particular case.

It could very well be however that this process in keras has changed. As
I said, I haven't followed keras development.


You may try it without this line, and if it works, I would merge the
respective patch :)



Otherwise: If you really need to understand that line perfectly I will
have a look at it, but not right now, as my master thesis is due in 4
days :|



Best,

Florian

**Comment by hoangcuong2011** [comment]
Thanks Florian for your quick response!

I don't know what your master thesis is about. But given your deep understanding of this advanced topic, I know the thesis is really awesome! :-P

Good luck!

Best,

Cuong.
